### ISSUE TITLE
Extension does not support unquoted values allowed in 0.12

### ISSUE THREAD

**Issue by kohenchia** [hashicorp/vscode-terraform#180]
The VSCode extension gives me an error for the following syntax:
```
variable foo {
  type = string
         ^^^^^^
```
It expects `"string"`, but the [unquoted form is supported in the official docs for 0.12](https://www.terraform.io/docs/configuration/variables.html).

The extension also gives me an error for the following syntax:
```
locals {
  xyz = "abc"
}

module "src" {
  name = local.xyz
         ^^^^^^^^^
} 
```
It expects the quoted form `"local.xyz"`, but the [unquoted form is supported in the official docs for 0.12](https://www.terraform.io/docs/configuration/locals.html).

**Comment by MattFenner** [comment]
I might have a go at the first part tomorrow (supporting unquoted types). I have never contributed to open source before, so it may take me a little bit to figure it out.

Any advice is appreciated.

**Comment by mauve** [comment]
the plugin does not yet support 0.12 syntax correctly, both the
tmLanguage files need updating as well as the indexing/parsing, refer
to https://github.com/mauve/vscode-terraform/issues/157 for
information.

**Comment by MattFenner** [comment]
FYI I have updated the tmLanguage files so at least the highlighting looks right. You will still get errors until #157 is fixed though.

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
